rdrf/rdrf/features/steps.py

```diff
@@ -7,17 +7,92 @@
 
 from . import utils
 
+from collections import OrderedDict
+import time
 
 logger = logging.getLogger(__name__)
 
 # Clearing all the aloe step definitions before we register our own.
 STEP_REGISTRY.clear()
 
 
+@step('I try to log in')
+def try_to_login(step):
+    world.browser.get(world.site_url + "login?next=/router/")
+
+
+@step('I try to register as an "([^"]+)" user')
+def try_to_register(step, registry):
+    registry_code = ''
+
+    if registry == 'Angelman':
+        registry_code = 'ang'
+
+    world.browser.get(world.site_url + registry_code + "/register")
+
+    # Plain text field parameters
+    params = OrderedDict([
+        ('id_username', 'sample@email.com'),
+        ('id_password1', 'admin123'),
+        ('id_password2', 'admin123'),
+        ('id_parent_guardian_first_name', 'John'),
+        ('id_parent_guardian_last_name', 'Smith'),
+        ('id_parent_guardian_date_of_birth', '1980-09-01'),
+        # Gender radio button
+        ('id_parent_guardian_address', 'Australia'),
+        ('id_parent_guardian_suburb', 'Australia'),
+        # Country dropdown
+        # State dropdown
+        ('id_parent_guardian_postcode', '6000'),
+        ('id_parent_guardian_phone', '98765432')
+    ])
+
+    # Populate plain text fields
+    for key, value in params.items():
+        # print("{0} {1}".format(key, value))
+        world.browser.find_element_by_id(key).send_keys(value + Keys.TAB)
+
+    # Select the gender radio button
+    # Male - 1, Female - 2, Indeterminate - 3
+    world.browser.find_element_by_css_selector("input[type='radio'][value='1']").click()
+
+    # Select the country and state dropdowns
+    world.browser.find_element_by_xpath("//select[@name='parent_guardian_country']/option[text()='Australia']").click()
+    world.browser.find_element_by_xpath("//select[@name='parent_guardian_state']/option[text()='Western Australia']").click()
+
+    # Fill out the patient details
+    world.browser.find_element_by_id('ui-id-2').click()
+
+    patient_params = OrderedDict([
+        ('id_first_name', 'Patient_First'),
+        ('id_surname', 'Patient_Surname'),
+        ('id_date_of_birth', '1985-01-01'),
+        # There is a gender field here
+        # Tick the "same details" box here
+    ])
+
+    for key, value in patient_params.items():
+        world.browser.find_element_by_id(key).send_keys(value + Keys.TAB)
+
+    radio = world.browser.find_element_by_id('id_gender')
+    world.browser.execute_script("arguments[0].click();", radio)
+
+    world.browser.find_element_by_id('same_address').send_keys(Keys.SPACE)
+
+    captcha_iframe_element = world.browser.find_element_by_xpath("//iframe[@title='recaptcha widget']")
+    world.browser.switch_to.frame(captcha_iframe_element)
+    captcha_element = world.browser.find_element_by_id('recaptcha-anchor').send_keys(Keys.SPACE)
+
+    time.sleep(4)
+    world.browser.switch_to_default_content()
+    world.browser.find_element_by_id('registration-submit').click()
```

Comment:
This submits the registration form, but there should be an assertion that the success message appears. I would put this as a subsequent step: I see the message "Your have successfully registered" ( or whatever the message is)